docs/sources/alerting/fundamentals/annotation-label/how-to-use-labels.md

```diff
@@ -23,6 +23,20 @@ This topic explains why labels are a fundamental component of alerting.
 
 {{< figure src="/static/img/docs/alerting/unified/rule-edit-details-8-0.png" max-width="550px" caption="Alert details" >}}
 
+# External Alertmanager Compatibility
+
+Grafana's built-in Alertmanager supports both Unicode label keys and values; however, if using an external Prometheus Alertmanager label keys must be compatible with their [data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels).
+This means label keys must only contain **ASCII letters**, **numbers**, as well as **underscores** and match the regex `[a-zA-Z_][a-zA-Z0-9_]*`.
+Any invalid characters will be removed or replaced by the Grafana alerting engine before being sent to the external Alertmanager according to the following rules:
+
+- `Whitespace` will be removed.
+- `ASCII characters` will be replaced with `_`.
+- `All other characters` will be replaced with their lower-case hex representation. If this is the first character it will be prefixed with `_`.
```

Comment:
I think we should move "If this is the first character it will be prefixed with `_`." to the same level with other rules because we add `_` to not just sanitized unicode characters but to strings that starts with a number.